utils.c

```diff
@@ -855,12 +855,14 @@ char *urlencode(const char * const str) {
 
 char *printmem(const char * const src, const size_t len, const int bitwidth) {
 	char *tmp;
+	uint8_t val;
 	size_t i;
 
 	tmp = zmalloc(2*len+1);
 	for (i = 0; i < len; ++i) {
-		tmp[i*2] = hextab[((uint8_t)src[i] ^ (uint8_t)(7-bitwidth)) >> 4];
-		tmp[i*2+1] = hextab[(src[i] ^ (uint8_t)(7-bitwidth)) & 0x0F];
+		val = (uint8_t)src[i] & (0xFF >> (8-bitwidth));
```

Comment:
Maybe it's better to declare `val` inside the for loop to make Codacy happy?

Reply:
Agreed, refactored as requested.